planner/mapper/mapper.go

```diff
@@ -119,36 +120,91 @@ func resolveOrderDependencies(
 
 	// If there is orderby, and any one of the condition fields that are join fields and have not been
 	// requested, we need to map them here.
+outer:
 	for _, condition := range source.Value().Conditions {
-		if len(condition.Fields) <= 1 {
-			continue
-		}
-
-		joinField := condition.Fields[0]
+		fields := condition.Fields[:] // copy slice
+		for {
+			numFields := len(fields)
+			if numFields == 2 {
```

Comment:
todo: Please document as to why toggling on `numFields == 2`, ` > 2` and `<= 1`, as it is not immediately clear to me

Reply:
Added comment